sklearn/utils/sparsefuncs_fast.pyx

```diff
@@ -184,7 +193,10 @@ def _csc_mean_variance_axis0(np.ndarray[floating, ndim=1] X_data,
         counts = endptr - startptr
 
         for j in xrange(startptr, endptr):
+            if X_data[j] == np.nan:
```

Comment:
Comparison to nan doesn't work. You need to use isnan, or look for a C implementation to use in Cython